File: Algorithms/1122.数组的相对排序/relativeSortArray.java
// 官方题解：计数排序
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int upper = 0;
        for (int x : arr1)
            upper = Math.max(upper, x);
        int[] frequency = new int[upper + 1];
        for (int x : arr1)
            ++frequency[x];
        int[] ans = new int[arr1.length];
        int index = 0;
        for (int x : arr2) {
            for (int i = 0; i < frequency[x]; ++i)
                ans[index++] = x;
            frequency[x] = 0;
        }
        for (int x = 0; x <= upper; ++x)
            for (int i = 0; i < frequency[x]; ++i)
                ans[index++] = x;
        return ans;
    }
}

File: Algorithms/1122/relativeSortArray.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        position = {x: i for i, x in enumerate(arr2)}
        arr1.sort(key=lambda x: position.get(x, 9999 + x))
        return arr1
